### PR TITLE
Adding Docker Compose for Multi Container Local Deployment

### PR DESCRIPTION
**Pull request by akroon3r**
# Description

This PR includes the following proposed change(s):

N/A

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Refactoring / Documentation
- [ ] Version change

if your change is a breaking change, please add `breaking change` label to this PR

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

## Does the change impact or break the Docker build?

- [x] Yes
- [ ] No

If Yes: Has Docker been updated accordingly?

- [x] Yes
- [ ] No

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: .dockerignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+/node_modules
+/.vscode
```
Comment (akroon3r) [inline .dockerignore:2]:
```suggestion
/.vscode

```

File: docker.env [hunk 2]
```diff
@@ -0,0 +1,6 @@
+POSTGRES_PASSWORD=snowball
+POSTGRES_DB=slam-db
+POSTGRES_USER=postgres
+POSTGRES_HOST=postgres
+POSTGRES_SYNCHRONIZE=true
+POSTGRES_PORT=5432
```
Comment (akroon3r) [inline docker.env:6]:
```suggestion
POSTGRES_PORT=5432

```

**Comment by akroon3r** [comment]
Please check this branch out and pull down the changes. The README.md should outline the steps required to get the containers off the ground.

Please reach out via teams so I can support as necessary.

File: src/app.module.ts [hunk 3]
```diff
@@ -10,23 +10,23 @@ import { SoftwareTitleEntity } from './software/software.entity';
...
     AssignedLicenseModule,
     LicenseeModule,
-    ConfigModule.forRoot(),
+    ConfigModule.forRoot({}),
     SoftwareModule,
     TypeOrmModule.forRoot({
       type: 'postgres',
       host: env.host,
-      port: env.port,
+      port: 5432,
```
Comment (akroon3r) [inline src/app.module.ts:29]:
```suggestion
      port: env.POSTGRES_PORT,
```

**Review by SDToews — APPROVED** [review]

**Pull request merged**